### PR TITLE
#535 | Transactions Route & Table

### PR DESCRIPTION
**Pull request by donnyquixotic**
# See #535 

## Description
Redesign of transaction table UI & functionality, update of route, and refactor to setup syntax for several components

## Test scenarios
- go to https://deploy-preview-546--teloscan-redesign.netlify.app/txs
    - you should see the transactions table
    - addresses in the from/to column should have a contract icon next to them if they are contracts
- click the eye icon in various rows with different qualities (e.g. is TLOS transfer, is Native deposit/withdraw, action has known name, action has no known name)
    - the tx preview should show and the values displayed should make sense for the row
- close the tx preview
- hover over a method name which has other instances on the table, e.g. "Transfer TLOS"
    - other method badges of the same kind should highlight
- hover over a `to` address which shows in the table more than once
    - other instances of the same address should highlight
- hover over a `from` address which shows in the table more than once
    - other instances of the same address should highlight
- click the copy button next to an address
    - the address should be copied to clipboard
- click the 'age' header
    - time format should switch
- click the 'Txn fee' header
    - the data should switch to show gas price
- try filtering by a block:  https://deploy-preview-546--teloscan-redesign.netlify.app/txs?block=286084888
    - data should be filtered
    - you should see appropriate labels in the page header
- try filtering by an account: https://deploy-preview-546--teloscan-redesign.netlify.app/txs?a=0xF5E8A439C599205C1aB06b535DE46681Aed1007a
    - data should be filtered
    - you should see appropriate labels in the page header
- try filtering by block and account: https://deploy-preview-546--teloscan-redesign.netlify.app/txs?a=0xF5E8A439C599205C1aB06b535DE46681Aed1007a&block=286084888
    - data should be filtered
    - you should see appropriate labels in the page header

## Checklist:

-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have cleaned up the code in the areas my change touches
-   [ ] My changes generate no new warnings
-   [ ] Any dependent changes have been merged and published in downstream modules
-   [ ] I have checked my code and corrected any misspellings
-   [ ] I have removed any unnecessary console messages
-   [ ] I have included all english text to the translation file and/or created a new issue with the required translations for the currently supported languages
-   [ ] I have tested for mobile functionality and responsiveness
-   [ ] I have added appropriate test coverage 

TODO:
- [x] action badge text should not be gray for unknown actions
- [x] change eye icon to use OutlineButton
- [x] change styling of transaction quick view status
- [x] in tx quick view, it should not say "Transfer for 0x0 TLOS from ... to ..." if it wasnt a transfer (i.e. if 0 value transaction), instead show action name
- [x] numerical values in quick view should not be hex
- [x] tx value in table should be formatted with toLocaleString
- [x] action badge should still highlight on hover for unknown actions
- [x] page should not say "for block xxxxx" if not filtered by block; filtering by block should work
- [x] action badge should not show an icon for deposits
- [x] clicking the question mark <th> causes the table to change size
- [x] i18n
- [x] enable filter by address

### CODE REVIEW

File: src/components/header/AppHeaderWallet.vue [hunk 1]
```diff
@@ -98,9 +104,10 @@ function copyAddress() {
 
 async function fetchUserBalance() {
     const response = await indexerApi.get(
-        `/account/${address.value}/balances?contract=___NATIVE_CURRENCY___&includeAbi=true`,
+        `/account/${address.value}/balances`,
     );
-    userSystemTokenBalanceWei.value = response.data?.results?.[0]?.balance ?? '0';
+    const tlos = response.data?.results?.find(({ contract }: { contract: string }) => contract === '___NATIVE_CURRENCY___');
```
Comment (ezra-sg) [inline src/components/header/AppHeaderWallet.vue:109]: fix for balance not showing correctly

File: src/components/ToolTip.vue [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+<q-tooltip class="tooltip">
+    <pre class="tooltip__content">{{ content }}</pre>
+</q-tooltip>
+</template>
+
+<style lang="sass" scoped>
+.tooltip
+    word-break: break-all
+    max-height: 100% !important
+    &_content
```
Comment (Viterbo) [inline src/components/ToolTip.vue:20]: This rule will not apply to the class `tooltip__content`, correct?
Also, it is scoped and uses SASS. Maybe, it's worth rewriting the entire style section.

```suggestion
    &__content
```

File: src/i18n/es-es/index.js [hunk 3]
```diff
@@ -10,6 +10,29 @@ export default {
...
+            status_label: 'Estado',
+            to: 'a',
+            transaction_action_label: 'Acción de transacción',
+            transaction_fee_label: 'Tarifa de transacción',
+            transactions: 'Transacciones',
+            transfer_for_x_tlos_from: 'Transferencia de { amount } TLOS desde',
+            transfer_tlos_action_name: 'Transferir TLOS',
+            withdraw_action_name: 'Retirar',
+            withdraw_native: 'Retirar (nativo)',
+            x_gas_used_of_y_limit: '{ amount } gas utilizado de un límite de { limit }',
```
Comment (Viterbo) [inline src/i18n/es-es/index.js:34]:
```suggestion
            x_gas_used_of_y_limit: '{ amount } de gas utilizado de un límite de { limit }',
```

File: src/i18n/pt-br/index.js [hunk 4]
```diff
@@ -10,6 +10,29 @@ export default {
...
+            status_label: 'Status',
+            to: 'para',
+            transaction_action_label: 'Ação da Transação',
+            transaction_fee_label: 'Taxa de Transação',
+            transactions: 'Transações',
+            transfer_for_x_tlos_from: 'Transferir { amount } TLOS de',
+            transfer_tlos_action_name: 'Transferir TLOS',
+            withdraw_action_name: 'Saque',
+            withdraw_native: 'Saque (nativo)',
+            x_gas_used_of_y_limit: '{ amount } gás usado de um limite de { limit }',
```
Comment (Viterbo) [inline src/i18n/pt-br/index.js:34]:
```suggestion
            x_gas_used_of_y_limit: '{ amount } de gás usado de um limite de { limit }',
```

**Review by Viterbo — APPROVED** [review]
Awesome work !!!
Just a few minor comments

**Comment by pmjanus** [comment]
Nice
- [ ] Nav bar overlapping Transactions header on mobile sizes.
- [ ] missing page "p" query parameter
- [ ] let's add a visual effect to the "eyeball" when modal is open to help identify which txn additional information it belongs to (let's talk about this modals movement when a user scrolls)
- [ ] can we limit to display the last 500k records and add the disclaimer on top of the table
- [ ] we dont have to have TLOS on Txn Fee since we state it is denominated in TLOS on the tooltip, we could do Txn Fee (TLOS) ?

File: src/components/TransactionDialog.vue [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+
+    try {
+        return gas.toNumber().toLocaleString(locale);
+    } catch (e) {
+        console.error(e);
+        return gas.toString();
+    }
+});
+const gasUsed = computed(() => {
+    const gas = BigNumber.from(props.trx.gasUsed ?? props.trx.gasused);
```
Comment (Viterbo) [inline src/components/TransactionDialog.vue:66]: Why is there this fallback? which source returns `gasused `instead `gasUsed`?

Comment (ezra-sg) [reply src/components/TransactionDialog.vue:66]: I don't know honestly, it was in the code when I took over. i'll check if it is needed

File: src/components/ToolTip.vue (hunk 2 above)
Comment (ezra-sg) [reply src/components/ToolTip.vue:20]: youre right, i will rewrite to match our standards 👍

File: src/components/TransactionDialog.vue (hunk 5 above)
Comment (ezra-sg) [reply src/components/TransactionDialog.vue:68]: yeah so the address/transactions endpoint has `gasused` and the normal transactions endpoint has `gasUsed`

**Pull request merged**